### PR TITLE
Mongo Crud without upsert.

### PR DESCRIPTION
**Pull request by setzer777**
Yesterday I finished up the transpiler, and starting the crud I got very emotional and the code ended up really fast haha there are still some coverage issues related with the number of branches tested (just 50%).

Just for the sanity of the PR checkers, I will include a better coverage in the next PR and I will include the upsert for the crud.

TODOs remaining:
1. Implement skip and limit for mongo crud.
2. Implement upsert in mongo crud.
3. Y pos ya, research more about a mongo connection pool.
4. Keep refactoring to beauty the code.

### CODE REVIEW

File: test/run/drivers/postgres/crud.spec.js [hunk 1]
```diff
@@ -26,7 +27,7 @@ describe('Postgres Crud', () => {
 
       beforeEach(done => {
         crud = model;
-        resetDatabase(['persons'])
+        resetDatabase(driver, ['persons'])
```
Comment (setzer777) [inline test/run/drivers/postgres/crud.spec.js]: Este pedo está bien culero, después haré un buildResetDatabase para mandarle el driver, que haga un closure acá tranquilón y ya nada más le mando las colecciones a droppear.

**Comment by codingpains** [comment]
- Fix Mongo Connection
- Extended Model Bitch

File: lib/drivers/mongo/crud.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+const MongoTranspiler = require('./transpiler');
+const buildMongoExecute = require('./build-mongo-execute');
+const {CantInsertRecordError} = require('./../../errors');
+
+const MongoCrud = (engine, model) => {
+  const transpiler = MongoTranspiler();
+  const collection = model.collection;
+
+  const count = query => {
+    return _executeCount(query);
```
Comment (codingpains) [inline lib/drivers/mongo/crud.js]: Never gonna give you up! Never gonna let you down!

File: lib/drivers/mongo/build-mongo-execute.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        if (error) return reject(error);
+        resolve(count);
+      });
+    });
+  };
+
+  const update = collection => {
+    return new Promise((resolve, reject) => {
+      const {query, update} = mongoQuery;
+      db.collection(collection).findAndModify(query, null, update, {new: true}, (error, result) => {
```
Comment (codingpains) [inline lib/drivers/mongo/build-mongo-execute.js]: Absstrract

**Pull request merged**